querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java

```diff
@@ -471,7 +471,19 @@ void serializeForDelete(QueryMetadata metadata, RelationalPath<?> entity) {
         dmlWithSchema = false;
 
         if (metadata.getWhere() != null) {
-            append(templates.getWhere()).handle(metadata.getWhere());
+            serializeForWhere(metadata);
+        }
+    }
+
+    void serializeForWhere(QueryMetadata metadata) {
+        boolean requireSchemaInWhere = templates.isRequiresSchemaInWhere();
+        if (requireSchemaInWhere) {
+            dmlWithSchema = true;
+        }
+        append(templates.getWhere()).handle(metadata.getWhere());
+
+        if (requireSchemaInWhere) {
+            dmlWithSchema = false;
```

Comment:
It looks like that member is only changed in that class. There is a `setDmlWithSchema` but I couldn't find any references.

I'm happy to do the new PR as well if you like.